### PR TITLE
chore: Defines internal guidelines for new changelog process

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-241384

Leaving link to other provider changelog guidelines as reference:

- AWS: [hashicorp/terraform-provider-aws/blob/main/docs/changelog-process.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/changelog-process.md)
- GCP: [magic-modules/contribute/release-notes/#type-specific-guidelines-and-examples](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/#type-specific-guidelines-and-examples)

HashiCorp guidelines: [hashicorp.com/terraform/plugin/best-practices/versioning#changelog-specification](https://developer.hashicorp.com/terraform/plugin/best-practices/versioning#changelog-specification)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by EspenAlbert — APPROVED** [review]
Great job 👏

File: contributing/changelog-process.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+### Header and entry values
+
+``````markdown
+```release-note:{HEADER}
+{ENTRY}
+```
+``````
+
+ `HEADER`
+ - Must be one of the following values: `breaking-change`, `new-resource`, `new-data-source`, `new-guide`, `note`, `enhancement`, `bug`. Examples for each type can be seen below.
```
Comment (lantoli) [inline contributing/changelog-process.md:49]: knit: i suppose most of the times new-resource and new-data-source will be in the same PR, don't know if new-resource would be enough to apply to either resource or ds so we don't need to have 2 (or 3 for plural ds) entries for a new resource/ds PR.

EDIT: i suppose it's not bad to have this so it's clear what's added:

release-note:new-data-resource
mongodbatlas_project

release-note:new-data-source
mongodbatlas_project

release-note:new-data-source
mongodbatlas_projects

**Review by lantoli — APPROVED** [review]

File: contributing/changelog-process.md (hunk 1 above)
Comment (AgustinBettati) [reply contributing/changelog-process.md:49]: yes, while more verbose I believe it is more explicit and removes any confusion

File: contributing/changelog-process.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+We use the [go-changelog](https://github.com/hashicorp/go-changelog) to generate and update the changelog from files created in the `.changelog/` directory. It is important that when you raise your Pull Request, there is a changelog entry which describes the changes your contribution makes. Not all changes require an entry in the changelog, guidance follows on what changes do.
+
+`@mongodb/docs-cloud-team` will be required reviewers for new changelog entry files contained in a Pull Request. We will wait up to 24 hours for a review, and after that proceed with the merge. Exceptions for merging ahead of the 24 hours may also apply.
+
+## Changelog format
+
+The changelog format requires an entry in the following format, where HEADER corresponds to the changelog category, and the entry is the changelog entry itself. The entry should be included in a file in the `.changelog` directory with the naming convention `{PR-NUMBER}.txt`. For example, to create a changelog entry for pull request 1234, there should be a file named `.changelog/1234.txt`.
```
Comment (maastha) [inline contributing/changelog-process.md:15]: [q] Sorry if I missed this, but how does cleanup of these entries work?

Comment (AgustinBettati) [reply contributing/changelog-process.md:15]: good observation, there is no clean up process in place for the `.changelog` directory files. From what I have seen in other providers they simply preserve them, or create an archive directory to place older files https://github.com/hashicorp/terraform-provider-aws/pull/33659. If we do reach a limitation or concern with amount of files, I see no concern in bulk removing old entries as the changelog generation process only looks for entries from the latest release onwards.

**Review by oarbusi — APPROVED** [review]
LGTM

**Pull request merged**